### PR TITLE
Update dependencies, fix vulnerabilities

### PR DESCRIPTION
**Pull request by swiftone**
Transferred from #132 to pass CI tests

### CODE REVIEW

**Comment by mraible** [comment]
@swiftone Can you add https://github.com/mraible/samples-js-react/commit/0f2148e479e523a3f672e7969082d558523b26ac to this PR? It upgrades to Okta React 3.0.7.

**Comment by mraible** [comment]
Doh! Need this one too: https://github.com/mraible/samples-js-react/commit/3d3e679d8f99cb0153e817f690a3edd172edd8b7

**Comment by swiftone** [comment]
@mraible - I just grabbed the latest of your branch on your fork into this one, does that suffice?

**Comment by mraible** [comment]
Yep!

**Comment by mraible** [comment]
Can I get a review and merge, please?

**Review by shuowu-okta — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,10 +4,21 @@ This repository contains several sample applications that demonstrate various Ok
...
 
-- **Client Id** - The client ID of the SPA application that you created earlier. This can be found on the "General" tab of an application, or the list of applications.  This identifies the application that tokens will be minted for.
-- **Issuer** - This is the URL of the authorization server that will perform authentication.  All Developer Accounts have a "default" authorization server.  The issuer is a combination of your Org URL (found in the upper right of the console home page) and `/oauth2/default`. For example, `https://dev-1234.oktapreview.com/oauth2/default`.
+> **Tip**: You can also create an account using the [Okta CLI](https://github.com/oktadeveloper/okta-cli) and `okta register`. To create an app, run `okta apps create` and use the settings below.
+
+Register your application by selecting **Applications** > **Add Application**. On the next screen, choose **Single Page App** and click **Next**.
+
+On the following screen, edit the application settings. For React applications running in developer mode, the port number should be 3000. Configure your app as follows:
+
+* **Base URI**: `http://localhost:3000`
```
Comment (aarongranick-okta) [inline README.md:15]: All the URLs configured in these samples seem to be using port 8080. Is this text about port 3000 accurate?

**Review by aarongranick-okta — CHANGES_REQUESTED** [review]
Please confirm port number. Although `create-react-app` uses a default port of 3000, I believe all of our samples and test harness apps are using a default port of 8080. This helps us switch easily between samples and test apps while using the same org config.

**Comment by mraible** [comment]
My philosophy is this: if someone creates a new React app, it'll use port 3000. If they create a new Angular app, it'll use 4200. If they create a new Vue app, it'll use 8080. 

If they clone this example and run `npm start`, it'll use 3000 by default. That is, unless we modify the default Create React App set up to use 8080 by default. I changed all the samples to use the ports developers, like myself, expect them to use. If they create a `testenv` file and want to run e2e tests, they'll need to add redirect URIs for 8080, but I think most developers will run the samples, they won't run `npm test` on them.

**Comment by aarongranick-okta** [comment]
> If they clone this example and run `npm start`, it'll use 3000 by default. 

We are explicitly setting the PORT environment variable here: https://github.com/okta/samples-js-react/blob/8b5b8c3c48915f120b570ca7ceb3e0e623a0a731/okta-hosted-login/package.json#L19

If they clone this repo and run `npm start` in one of the sample directories it will use 8080 by default.

When investigating customer issues we switch between samples and test apps A LOT. We will typically use the same org and testenv file across all these apps, which saves us time and reduces human error. It would be very inconvenient to require different testenv configurations between frameworks.

**Comment by mraible** [comment]
OK. That's this samples repo. If we re-worked it to be like the Vue samples, it'd use Create React App's defaults and not have this in there, or we could modify the [default script to specify a port](https://stackoverflow.com/a/41770848/65681). If you'd prefer 8080, we should change the Angular samples to use it too. 

This line https://github.com/okta/samples-js-angular/blob/master/custom-login/package.json#L7, probably needs to change to be `ng start --port 8080` and the README needs to be updated too.

**Review by aarongranick-okta — APPROVED** [review]

**Pull request merged**